libqtile/backend/wayland/core.py

```diff
@@ -774,6 +790,19 @@ def stack_windows(self) -> None:
         else:
             self.stacked_windows = self.mapped_windows
 
+    def check_idle_inhibitor(self) -> None:
+        """
+        Checks if any window that is currently mapped has idle inhibitor
+        and if so inhibits idle
+        """
+
+        idle_inhibited = any(
+            win.is_idle_inhibited
+            for win in self.mapped_windows
+            if isinstance(win, window.Window) and not isinstance(win, (window.Internal, window.Static))
+        )
+        self.idle.set_enabled(self.seat, not idle_inhibited)
```

Comment:
I think the generator for loop hurts readability:
```suggestion
        for win in self.mapped_windows:
	    if isinstance(win, window.Window) and win.is_idle_inhibited:
		self.idle.set_enabled(self.seat, True)
		return

	self.idle.set_enabled(self.seat, False)
```